components/brave_wallet/browser/eth_response_parser.cc

```diff
@@ -15,6 +15,17 @@
 
 namespace brave_wallet {
 
+bool ParseStringResult(const std::string& json, std::string* value) {
+  DCHECK(value);
+
+  std::string result;
+  if (!ParseSingleStringResult(json, &result))
+    return false;
+
+  size_t offset = 2 /* len of "0x" */ + 64 /* len of offset to array */;
+  return brave_wallet::DecodeString(offset, result, value);
+}
+
```

Comment:
I created this function because the logic is common between `ParseEnsResolverContentHash`, `ParseUnstoppableDomainsProxyReaderGet`, and for parsing tokenURI responses.  I didn't end up using this in `ParseERC721TokenUri` though because `ParseStringResult` fetches the `result` key of the JSON in the RPC response and then RLP decodes it atomically.  I didn't want this to be atomic because I wanted to preserve the ability to pass an `kExceedsLimit` and other provider errors if the first step fails, and a `kParsingError` if the second step fails in `JsonRpcService::OnGetERC721TokenUri`.

Reply:
agreed in slack DM to use `ParseStringResult` instead.